### PR TITLE
Bridging to CoreFoundation types for InputStream/OutputStream

### PR DESCRIPTION
**Pull request by apocolipse**
Relies on #605 for fixes to spelling.

Adds proper bridging to `InputStream` and `OutputStream` to their CF counterparts.

### CODE REVIEW

**Comment by phausler** [comment]
This seems ok, but what is the exact use case here? the CF bridge protocols are internal only. Is this somehow intending to allow Foundation's OutputStream to be passed/fetched directly to/from CF as a CFWriteStream?

**Comment by apocolipse** [comment]
@phausler Yes, intending to allow direct access to CFWriteStream (and read), Currently there's a discrepancy between CFStreamPropertyKeys and Stream.PropertyKeys such that if you want to set specific keys it must be done on a CF stream, and vice versa.  i.e. Stream.PropertyKey.socketSecuritylevelKey is implemented but not for CFStreams.  As well as things like CFStreamCreatePairWithSocketToHost  are not implemented for Stream (yet).

Ideally, all of the Stream.PropertyKeys should eventually align with all kCFStreamPropertyKeys, and utilities like getStreamsToHost in Stream will be fully implemented (#588 #589) so that lower level CFStream , but for now bridging helps get around some of the current inconsistencies.

**Comment by apocolipse** [comment]
@phausler are you indicating this won't allow for external bridging?  If not, is there any way that will?  Otherwise I'd opt to just expose the internal `_cfObject` and `init(reference:)`

**Comment by phausler** [comment]
CF is an internal implementation detail of swift-corelibs-foundation. I often attempt to dissuade Swift developers from using CF - because it is quite painful to deal with typing unsafe so many times and often the Foundation counterparts are faster anyhow. 

In this particular case I was curious if the intent is to directly allow APIs from CF to pass transparently across to Foundation - if that is the case then we need to make certain we have a pattern more similar to NSCFString and such to do toll free bridging. What you have here is a "has-a" pattern bridge, I was wanting to make certain you were not intending for it to be a "is-a" pattern bridge.

tl;dr To hand out a CFWriteStream and allow use as a NSStream it needs to be the "is-a" pattern.

**Comment by parkera** [comment]
I'm going to go ahead and close this for now. Please re-open when changes requested are made. Thanks.
